clocklogger/input.py

```diff
@@ -92,9 +92,7 @@ def get_samples(self, num_samples):
 
     def consume(self, num_samples):
         """Mark num_samples as having been used"""
-        num_to_read = num_samples - self.buffer.shape[0]
-        if num_to_read > 0:
-            self.get_samples(num_to_read)
+        self.get_samples(num_samples)
```

Comment:
Note for my benefit as I couldn't see the problem straight away -- `self.get_samples` accepts `num_samples` not `num_to_read` so the calculation was duplicated